From 6e971937c4306af93bb145f70662e4926deda8e0 Mon Sep 17 00:00:00 2001
From: Laurent Pinchart <laurent.pinchart+renesas@ideasonboard.com>
Date: Thu, 28 Mar 2013 13:10:31 +0100
Subject: [PATCH 0993/1083] drm/rcar-du: Add XRGB1555 format support

Signed-off-by: Laurent Pinchart <laurent.pinchart+renesas@ideasonboard.com>
(cherry picked from commit 9ae34613ae7e012d26958faa2b0f1c2fd79c9b24)

Signed-off-by: Simon Horman <horms+renesas@verge.net.au>
(cherry picked from commit 091168722e21846a241a7ae28f3d6a62a29d3f1a)

Signed-off-by: Ryo Kataoka <ryo.kataoka.wt@renesas.com>
---
 drivers/gpu/drm/rcar-du/rcar_du_kms.c   | 6 ++++++
 drivers/gpu/drm/rcar-du/rcar_du_plane.c | 2 ++
 2 files changed, 8 insertions(+)

diff --git a/drivers/gpu/drm/rcar-du/rcar_du_kms.c b/drivers/gpu/drm/rcar-du/rcar_du_kms.c
index b6cd53f..2c537c2 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_kms.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_kms.c
@@ -42,6 +42,12 @@ static const struct rcar_du_format_info rcar_du_format_infos[] = {
 		.pnmr = PnMR_SPIM_ALP | PnMR_DDDF_ARGB,
 		.edf = PnDDCR4_EDF_NONE,
 	}, {
+		.fourcc = DRM_FORMAT_XRGB1555,
+		.bpp = 16,
+		.planes = 1,
+		.pnmr = PnMR_SPIM_TP | PnMR_DDDF_ARGB,
+		.edf = PnDDCR4_EDF_NONE,
+	}, {
 		.fourcc = DRM_FORMAT_XRGB8888,
 		.bpp = 32,
 		.planes = 1,
diff --git a/drivers/gpu/drm/rcar-du/rcar_du_plane.c b/drivers/gpu/drm/rcar-du/rcar_du_plane.c
index 75197ad..e1e0a9b 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_plane.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_plane.c
@@ -161,6 +161,7 @@ static void rcar_du_plane_setup_mode(struct rcar_du_plane *plane,
 		break;
 
 	case DRM_FORMAT_ARGB1555:
+	case DRM_FORMAT_XRGB1555:
 		colorkey = ((plane->colorkey & 0xf80000) >> 9)
 			 | ((plane->colorkey & 0x00f800) >> 6)
 			 | ((plane->colorkey & 0x0000f8) >> 3);
@@ -391,6 +392,7 @@ static const struct drm_plane_funcs rcar_du_plane_funcs = {
 static const uint32_t formats[] = {
 	DRM_FORMAT_RGB565,
 	DRM_FORMAT_ARGB1555,
+	DRM_FORMAT_XRGB1555,
 	DRM_FORMAT_XRGB8888,
 	DRM_FORMAT_ARGB8888,
 	DRM_FORMAT_UYVY,
-- 
1.8.3.2

